app/services/datetime_parser.rb

```diff
@@ -6,34 +6,45 @@ def initialize(date:, time:)
     @raw_time = time.to_s
   end
 
+  def issues
+    @issue_items
+  end
+
+  def issues_for(param)
+    @issue_items.filter { |issue| issue.keys.first == param }.map { |issue| issue.values.first }
```

Comment:
The `@` isn't needed here.

Reply:
Are you trying to get the value for the first issue of a specific type? Or are you trying to get them all?

If you're want them all, you should be able to do `issue.items.pluck(param)` to get them all and `issue.items.pluck(param).first` to get the first one.

Reply:
Don't think that works. `issue_items` at this point looks something like this:

```ruby
issue_items = [
  { schedule_date: "Invalid date" },
  { schedule_date: "Date missing!" },
  { schedule_time: "Invalid time!" }
]
```

Reply:
So what are you trying to do here?

Reply:
`issue_items.pluck(:schedule_date)` will return `["Invalid date", "Date missing!"]`

Reply:
Hmm I think the issue was `pluck` would return `[ nil ]` if nothing found, as opposed to `[]`. Have now changed to `issue_items.pluck(param).compact` 👍